### PR TITLE
removed variants from intervar run that got a call from submission files

### PR DESCRIPTION
**Pull request by naqvia**
Related to https://github.com/diskin-lab-chop/AutoGVP/issues/50

### CODE REVIEW

File: AutoGVP/01-annotate_variants_custom_input.R [hunk 1]
```diff
@@ -134,14 +134,14 @@ address_conflicting_intrep <- function(clinvar_anno_vcf_df)
...
-vcf_df  <-  vroom(as.character(input_vcf_file), comment = "#",delim="\t", col_names = c("CHROM","POS","ID","REF","ALT","QUAL","FILTER","INFO","FORMAT","Sample"), trim_ws = TRUE, show_col_types = FALSE) %>%
+vcf_df  <-  vroom(input_clinVar_file, comment = "#",delim="\t", col_names = c("CHROM","POS","ID","REF","ALT","QUAL","FILTER","INFO","FORMAT","Sample"), trim_ws = TRUE, show_col_types = FALSE) %>%
             mutate(
               vcf_id= str_remove_all(paste (CHROM,"-",POS,"-",REF,"-",ALT), " "), 
               vcf_id = str_replace_all(vcf_id, "chr", "")
             )
 
 ## add clinvar table to this (INFO)
-clinvar_anno_vcf_df  <- vroom(input_clinVar_file, comment = "#", delim="\t", col_names = c("CHROM","POS","ID","REF","ALT","QUAL","FILTER","INFO"),trim_ws = TRUE, show_col_types = FALSE) %>%
+clinvar_anno_vcf_df  <- vroom("/Users/naqvia/Documents/GitHub/pathogenicity-assessment/AutoGVP/input/clinvar.vcf.gz", comment = "#", delim="\t", col_names = c("CHROM","POS","ID","REF","ALT","QUAL","FILTER","INFO"),trim_ws = TRUE, show_col_types = FALSE) %>%
```
Comment (jharenza) [inline AutoGVP/01-annotate_variants_custom_input.R:144]: Please fix this path

File: AutoGVP/01-annotate_variants_custom_input.R [hunk 2]
```diff
@@ -174,10 +174,10 @@ submission_summary_df <- vroom(input_submission_file, comment = "#",delim="\t",
...
                                              "SCV","SubmittedGeneSymbol","ExplanationOfInterpretation"),
                                show_col_types = FALSE) %>% dplyr::select("VariationID","ClinicalSignificance") %>% 
-  group_by(VariationID) %>%
-  arrange(ClinicalSignificance) %>%
-  dplyr::slice(1) %>%
-  ungroup
+                                group_by(VariationID) %>%
+                                arrange(DateLastEvaluated) %>%
+                                dplyr::slice(1) %>%
+                                ungroup
```
Comment (jharenza) [inline AutoGVP/01-annotate_variants_custom_input.R:180]:
```suggestion
                                ungroup()
```

File: AutoGVP/01-annotate_variants_custom_input.R [hunk 3]
```diff
@@ -188,7 +188,11 @@ submission_info_df  <-  vroom(input_variant_summary, delim="\t",
                             vcf_id= str_remove_all(paste (Chromosome,"-",PositionVCF,"-",ReferenceAlleleVCF,"-",AlternateAlleleVCF), " "),
                             vcf_id = str_replace_all(vcf_id, "chr", ""),
                             VariationID=as.double(noquote(VariationID))
-                            )
+                            ) %>% 
+                          group_by(VariationID) %>%
+                          arrange(LastEvaluated) %>%
+                          dplyr::slice(1) %>%
+                          ungroup
```
Comment (jharenza) [inline AutoGVP/01-annotate_variants_custom_input.R:195]:
```suggestion
                          ungroup()
```

File: AutoGVP/01-annotate_variants_custom_input.R [hunk 4]
```diff
@@ -202,11 +206,10 @@ entries_for_cc_in_submission <- inner_join(submission_final_df,entries_for_cc, b
 ## modified: any cases that do NOT have the B, LB, P, LP, VUS call must also go to intervar
 additional_intervar_cases <-  filter(clinvar_anno_vcf_df, final_call!="Benign",final_call!="Pathogenic", final_call != "Likely_benign",final_call!="Likely_pathogenic", final_call != "Uncertain_significance")
 
-
 clinvar_anti_join_vcf_df <- clinvar_anti_join_vcf_df %>% mutate(QUAL = as.character(QUAL))
 
 ## filter only those variant entries that need an InterVar run (No Star) and add the additional intervar cases from above
-entries_for_intervar <- filter(clinvar_anno_vcf_df, Stars == "0", na.rm = TRUE) %>% bind_rows((additional_intervar_cases)) %>% bind_rows(clinvar_anti_join_vcf_df)
+entries_for_intervar <- filter(clinvar_anno_vcf_df, Stars == "0", na.rm = TRUE) %>% bind_rows((additional_intervar_cases)) %>% bind_rows(clinvar_anti_join_vcf_df) #%>% anti_join(entries_for_cc_in_submission, by="vcf_id")
```
Comment (jharenza) [inline AutoGVP/01-annotate_variants_custom_input.R:212]: Remove commented text

File: AutoGVP/01-annotate_variants_custom_input.R [hunk 5]
```diff
@@ -274,8 +277,8 @@ autopvs1_results    <-  read_tsv(input_autopvs1_file, col_names = TRUE) %>%
   mutate(
     vcf_id = str_remove_all(paste (vcf_id), " "),
     vcf_id = str_replace_all(vcf_id, "chr", "")
-  ) %>%
-  dplyr::filter(vcf_id %in% clinvar_anno_intervar_vcf_df$vcf_id)
+  ) #%>%
+  #dplyr::filter(vcf_id %in% clinvar_anno_intervar_vcf_df$vcf_id)
```
Comment (jharenza) [inline AutoGVP/01-annotate_variants_custom_input.R:281]: Delete this

**Review by jharenza — CHANGES_REQUESTED** [review]
Untested, but left code change comments. I'm also going to work with Chris to implement the R styler package on this code so it adheres to tidyverse guidelines and is more readable. Stay tuned!

**Review by jungkim2 — APPROVED** [review]
tested the code, it solved duplicated variant output. But other issues have been identified. submitting separate issues for those.

**Comment by naqvia** [comment]
Current issue has been fixed. There are other issues that @jungkim2 will write an issue for!

**Review by jungkim2 — APPROVED** [review]

**Comment by naqvia** [comment]
latest commit is related to https://github.com/diskin-lab-chop/AutoGVP/issues/76

**Pull request merged**